components/automate-builder-api-proxy/habitat/config/nginx.conf

```diff
@@ -0,0 +1,143 @@
+daemon off;
+pid {{pkg.svc_var_path}}/pid;
+
+worker_processes  {{cfg.nginx.worker_processes}};
+worker_rlimit_nofile {{cfg.nginx.worker_rlimit_nofile}};
+
+events {
+  worker_connections  {{cfg.nginx.worker_connections}};
+}
+
+http {
+  client_body_temp_path {{pkg.svc_var_path}}/nginx/client-body;
+  fastcgi_temp_path {{pkg.svc_var_path}}/nginx/fastcgi;
+  proxy_temp_path {{pkg.svc_var_path}}/nginx/proxy;
+  scgi_temp_path {{pkg.svc_var_path}}/nginx/scgi_temp_path;
+  uwsgi_temp_path {{pkg.svc_var_path}}/nginx/uwsgi;
+
+  include        {{ pkgPathFor "core/nginx" }}/config/mime.types;
+  default_type   application/octet-stream;
+
+  sendfile       {{cfg.http.sendfile}};
+  tcp_nopush     {{cfg.http.tcp_nopush}};
+  tcp_nodelay    {{cfg.http.tcp_nodelay}};
+
+  keepalive_requests {{cfg.http.keepalive_requests}};
+  keepalive_timeout  {{cfg.http.keepalive_timeout}};
+
+ {{~#if cfg.nginx.enable_gzip}}
+  gzip on;
+  gzip_vary on;
+  gzip_min_length 256;
+  gzip_proxied expired no-cache no-store private auth;
+  gzip_types
+    application/javascript
+    application/json
+    application/vnd.ms-fontobject
+    application/x-font-ttf
+    font/opentype
+    image/svg+xml
+    image/x-icon
+    text/css;
+
+  gzip_disable "MSIE [1-6]\.";
+  {{~/if}}
+
+  open_file_cache max=1000 inactive=20s;
+  open_file_cache_valid 30s;
+  open_file_cache_min_uses 2;
+  open_file_cache_errors on;
+
+  add_header X-Frame-Options SAMEORIGIN;
+  add_header X-Content-Type-Options nosniff;
+  add_header X-XSS-Protection "1; mode=block";
+  add_header Alternate-Protocol  443:npn-spdy/3;
+
+  add_header "X-UA-Compatible" "IE=Edge";
+  add_header Strict-Transport-Security "max-age=31536000; includeSubDomains;";
+
+  {{~#if cfg.nginx.enable_caching}}
+  proxy_cache_path {{pkg.svc_var_path}}/cache levels=1:2 keys_zone=my_cache:10m max_size=10g inactive=60m use_temp_path=off;
+  proxy_cache_key "$scheme$proxy_host$uri$is_args$args $http_user_agent";
+  proxy_cache_valid 10m; # this only caches 200, 301, 302
+  {{~/if}}
+
+  log_format chef '$remote_user [$time_local]  '
+                  '"$request" $status "$request_time" $body_bytes_sent '
+                  '"$http_referer" "$http_user_agent" "$upstream_addr" "$upstream_status" "$upstream_response_time" $request_length';
+
+  upstream backend {
+    {{~#each bind.automate-builder-api.members as |member|}}
+    server {{member.sys.ip}}:{{member.cfg.http-port}};
+    {{~/each}}
+    keepalive {{cfg.http.keepalive_connections}};
+  }
+  server {
+    index /index.html;
+    server_name  {{sys.hostname}};
+    root         {{pkgPathFor "habitat/builder-api-proxy"}}/app;
+
+    listen {{cfg.service.host}}:{{cfg.service.port}};
+
+    access_log /dev/stdout chef;
+
+    #ssl on;
```

Comment:
Are these here for an eventual TODO?

Reply:
yea, unclear if we should bother with ssl. i don't think builder-api supports it